components/signup.php

```diff
@@ -0,0 +1,213 @@
+<?php
+/**
+ * Manage BuddyPress Signups.
+ *
+ * @since 1.5.0
+ */
+class BPCLI_Signup extends BPCLI_Component {
+
+	/**
+	 * Add a signup.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--user-login=<user-login>]
+	 * : User login for the signup. If none is provided, a random one will be used.
+	 *
+	 * [--user-email=<user-email>]
+	 * : User email for the signup. If none is provided, a random one will be used.
+	 *
+	 * [--meta=<meta>]
+	 * : User meta for the signup.
+	 *
+	 * [--silent=<silent>]
+	 * : Silent the signup creation.
+	 * ---
+	 * default: false.
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *
+	 *   $ wp bp signup add --user-login=test_user --user-email=teste@site.com
+	 *   $ wp bp signup add --user-login=test_user --user-email=teste@site.com --silent=1
+	 */
+	public function add( $args, $assoc_args ) {
+		$r = wp_parse_args( $assoc_args, array(
+			'user_login'     => '',
+			'user_email'     => '',
+			'activation_key' => wp_generate_password( 32, false ),
+			'meta'           => '',
+			'silent'         => false,
+		) );
+
+		// Add a random user login if none is provided.
+		if ( empty( $r['user_login'] ) ) {
+			$r['user_login'] = $this->get_random_login();
+		}
+
+		// Sanitize login (random or not).
+		$r['user_login'] = preg_replace( '/\s+/', '', sanitize_user( $r['user_login'], true ) );
+
+		// Add a random email if none is provided.
+		if ( empty( $r['user_email'] ) ) {
+			$r['user_email'] = is_email( $this->get_random_login() . '@site.com' );
+		}
+
+		// Sanitize email (random or not).
+		$r['user_email'] = sanitize_email( $r['user_email'] );
+
+		$id = BP_Signup::add( $r );
+
+		if ( $r['silent'] ) {
+			return;
+		}
+
+		if ( $id ) {
+			WP_CLI::success( sprintf( 'Successfully added new user signup (ID #%d)', $id ) );
+		} else {
+			WP_CLI::error( 'Could not add a user signup.' );
+		}
+	}
+
+	/**
+	 * Delete a signup.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <signup-id>
+	 * : Identifier for the signup.
+	 *
+	 * ## EXAMPLE
+	 *
+	 *   $ wp bp signup delete 520
+	 */
+	public function delete( $args, $assoc_args ) {
+		if ( BP_Signup::delete( $args[0] ) ) {
+			WP_CLI::success( 'Signup deleted.' );
+		} else {
+			WP_CLI::error( 'Could not delete signup.' );
+		}
+	}
+
+	/**
+	 * Activate a signup.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <activation-key>
+	 * : Identifier for the activation key.
+	 *
+	 * ## EXAMPLE
+	 *
+	 *   $ wp bp signup activate ee48ec319fef3nn4
+	 */
+	public function activate( $args, $assoc_args ) {
+
+		$id = bp_core_activate_signup( $args[0] );
+
+		if ( $id ) {
+			WP_CLI::success( sprintf( 'Signup activated, new user (ID #%d)', $id ) );
+		} else {
+			WP_CLI::error( 'Signup not activated.' );
+		}
+	}
+
+	/**
+	 * Generate random signups.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--count=<number>]
+	 * : How many signups to generate.
+	 * ---
+	 * default: 100
+	 * ---
+	 *
+	 * ## EXAMPLE
+	 *
+	 *  $ wp bp signup generate --count=50
+	 */
+	public function generate( $args, $assoc_args ) {
```

Comment:
@boonebgorges Should we drop the `--count` here? the convention is to keep it but I think we are safe to remove it.